File: 프로그래머스/JS/팩토리얼.js
/* 숫자 n이 있다고 하자. n을 1부터 ++1을 하는 와중에, n!를 하자.
그리고 처음으로 n보다 클 때 바로 return 0를 해서 멈추게끔 하자.
*/

function solution(n) {
    let i = 1;
    let factorial = 1;

    while (true) { 
        factorial *= i;
        if (factorial > n) {
            return i - 1; 
        }
        i++;
    }
}

File: 코테 실습/파이썬/팩토리얼.py
def solution(n):
    i = 1
    factorial = 1

    # 해결: while 루프와 그 안의 코드를 solution 함수 안에 **올바르게 들여쓰기**!
    while True:
        factorial *= i
        if factorial > n:
            return i - 1
        i += 1